geode-core/src/main/java/org/apache/geode/management/internal/beans/stats/GatewaySenderOverflowMonitor.java

```diff
@@ -26,64 +27,118 @@
 import org.apache.geode.internal.statistics.ValueMonitor;
 
 public class GatewaySenderOverflowMonitor extends MBeanStatsMonitor {
+  private Map<Statistics, ValueMonitor> monitors;
+  private Map<Statistics, StatisticsListener> listeners;
+  private AtomicLong lruEvictions = new AtomicLong(0);
+  private AtomicLong bytesOverflowedToDisk = new AtomicLong(0);
+  private AtomicLong entriesOverflowedToDisk = new AtomicLong(0);
 
-  private volatile long entriesOverflowedToDisk = 0;
+  public long getLruEvictions() {
+    return lruEvictions.get();
+  }
 
-  private volatile long bytesOverflowedToDisk = 0;
+  public long getBytesOverflowedToDisk() {
+    return bytesOverflowedToDisk.get();
+  }
 
-  private volatile long lruEvictions = 0;
+  public long getEntriesOverflowedToDisk() {
+    return entriesOverflowedToDisk.get();
+  }
 
-  private Map<Statistics, ValueMonitor> monitors;
+  Map<Statistics, ValueMonitor> getMonitors() {
+    return monitors;
+  }
 
-  private Map<Statistics, StatisticsListener> listeners;
+  Map<Statistics, StatisticsListener> getListeners() {
+    return listeners;
+  }
 
   public GatewaySenderOverflowMonitor(String name) {
     super(name);
     monitors = new HashMap<>();
     listeners = new HashMap<>();
   }
 
+  Number computeDelta(DefaultHashMap statsMap, String name, Number currentValue) {
+    if (name.equals(StatsKey.GATEWAYSENDER_LRU_EVICTIONS)) {
+      Number prevValue = statsMap.get(StatsKey.GATEWAYSENDER_LRU_EVICTIONS).longValue();
+      return currentValue.longValue() - prevValue.longValue();
+    }
+
+    if (name.equals(StatsKey.GATEWAYSENDER_ENTRIES_OVERFLOWED_TO_DISK)) {
+      Number prevValue =
+          statsMap.get(StatsKey.GATEWAYSENDER_ENTRIES_OVERFLOWED_TO_DISK).longValue();
+      return currentValue.longValue() - prevValue.longValue();
+    }
+
+    if (name.equals(StatsKey.GATEWAYSENDER_BYTES_OVERFLOWED_TO_DISK)) {
+      Number prevValue = statsMap.get(StatsKey.GATEWAYSENDER_BYTES_OVERFLOWED_TO_DISK).longValue();
+      return currentValue.longValue() - prevValue.longValue();
+    }
+
+    return 0;
+  }
+
+  void increaseStats(String name, Number value) {
+    if (name.equals(StatsKey.GATEWAYSENDER_LRU_EVICTIONS)) {
+      lruEvictions.set(lruEvictions.get() + value.longValue());
+      return;
+    }
+
+    if (name.equals(StatsKey.GATEWAYSENDER_ENTRIES_OVERFLOWED_TO_DISK)) {
+      entriesOverflowedToDisk.set(entriesOverflowedToDisk.get() + value.longValue());
+      return;
+    }
+
+    if (name.equals(StatsKey.GATEWAYSENDER_BYTES_OVERFLOWED_TO_DISK)) {
+      bytesOverflowedToDisk.set(bytesOverflowedToDisk.get() + value.longValue());
```

Comment:
use getAndAdd instead of set() and get()

Reply:
👍